src/OperationResponse.php

```diff
@@ -0,0 +1,230 @@
+        }
+        if (isset($options['metadataReturnType'])) {
+            $this->metadataReturnType = $options['metadataReturnType'];
+        }
+        if (isset($options['lastProtoResponse'])) {
+            $this->lastProtoResponse = $options['lastProtoResponse'];
+        }
+    }
+
+    /**
+     * Check whether the operation has completed.
+     *
+     * @return bool
+     */
+    public function isDone()
+    {
+         return (is_null($this->lastProtoResponse) || is_null($this->lastProtoResponse->getDone()))
+             ? false
+             : $this->lastProtoResponse->getDone();
+    }
+
+    /**
+     * Get the formatted name of the operation
+     *
+     * @return string The formatted name of the operation
+     */
+    public function getName()
+    {
+        return $this->operationName;
+    }
+
+    /**
+     * Poll the server in a loop until the operation is complete.
+     *
+     * Return true if the operation completed successfully, otherwise return
+     * false.
+     *
+     * The $pollSettings optional argument can be used to control the polling loop.
+     *
+     * @param array $options {
+     *                       Options for configuring the polling behaviour.
+     *
+     *     @type float $pollingIntervalSeconds The polling interval to use, in seconds.
+     *                                          Default: 1.0
+     * }
+     * @return boolean Indicates if the operation completed successfully.
+     */
+    public function pollUntilComplete($options = [])
+    {
+        $defaultPollSettings = [
+            'pollingIntervalSeconds' => $this::DEFAULT_POLLING_INTERVAL,
+        ];
+        $pollSettings = array_merge($defaultPollSettings, $options);
+
+        $pollingIntervalMicros = $pollSettings['pollingIntervalSeconds'] * 1000000;
+
+        while (!$this->isDone()) {
+            usleep($pollingIntervalMicros);
+            $this->reload();
+        }
+
+        return $this->lastProtoResponse->hasResponse();
+    }
+
+    /**
+     * Reload the status of the operation with a request to the service.
+     */
+    public function reload()
+    {
+        if ($this->deleted) {
+            throw new ValidationException("Cannot call reload() on a deleted operation");
+        }
+        $name = $this->getName();
+        $this->lastProtoResponse = $this->operationsClient->getOperation($name);
+    }
+
+    /**
+     * Return the result of the operation. If the operation is not complete, or if the operation
+     * failed, return null.
+     * @return mixed The result of the operation, or null if the operation failed or is not complete
+     */
+    public function getResult()
+    {
+        if (!$this->isDone() || !$this->lastProtoResponse->hasResponse()) {
+            return null;
+        }
+
+        $anyResponse = $this->lastProtoResponse->getResponse();
+        if (is_null($this->operationReturnType)) {
+            return $anyResponse;
```

Comment:
If `$this->operationReturnType` is defined, does it share the same way to interface with the code as `$anyResponse`?